$OpenBSD: patch-Makefile,v 1.1.1.1 2021/10/16 10:55:00 sthen Exp $

- don't force compiler flags
- don't unconditionally set -ftrampolines, it is gcc-only.
(Blocksruntime isn't working correctly on OpenBSD anyway so clang
is not yet an option, but patched away here to allow for easier
testing, and added-back in the port Makefile).
- fix basename(1) gnuism

Index: Makefile
--- Makefile.orig
+++ Makefile
@@ -139,7 +139,7 @@ ASTTOPDIR:=$(subst $(space),\$(space),$(CURDIR))
 OVERWRITE=y
 
 # Include debug and macro symbols in the executables (-g) and profiling info (-pg)
-DEBUG=-g3
+#DEBUG=-g3
 
 # Asterisk.conf is located in ASTETCDIR or by using the -C flag
 # when starting Asterisk
@@ -188,7 +188,9 @@ OPTIONS=
 
 ifeq ($(findstring -save-temps,$(_ASTCFLAGS) $(ASTCFLAGS)),)
   ifeq ($(findstring -pipe,$(_ASTCFLAGS) $(ASTCFLAGS)),)
+    ifneq ($(OSARCH),OpenBSD)
     _ASTCFLAGS+=-pipe
+    endif
   endif
 endif
 
@@ -232,7 +234,7 @@ ifeq ($(OSARCH),NetBSD)
 endif
 
 ifeq ($(OSARCH),OpenBSD)
-  _ASTCFLAGS+=-pthread -ftrampolines
+  _ASTCFLAGS+=-pthread
 endif
 
 ifeq ($(OSARCH),linux-uclibc)
@@ -493,7 +495,7 @@ doc/core-en_US.xml: makeopts .lastclean $(XML_core_en_
 			MODULEINFO=$$($(AWK) -f build_tools/get_moduleinfo $$i) ; \
 			if [ -n "$$MODULEINFO" ] ; \
 			then \
-				echo "<module language=\"en_US\" name=\"`$(BASENAME) -s .c $$i`\">" >> $@ ; \
+				echo "<module language=\"en_US\" name=\"`$(BASENAME) $$i .c`\">" >> $@ ; \
 				echo "$$MODULEINFO" >> $@ ; \
 				echo "</module>" >> $@ ; \
 			fi ; \
